api/src/api/taxFilingRouter.test.ts

```diff
@@ -102,6 +164,25 @@ describe("taxFilingRouter", () => {
       expect(response.status).toEqual(200);
     });
 
+    it("decrypts the taxId field using the encryptedTaxId field if it is masked", async () => {
+      const taxIdAndBusinessName = generateTaxIdAndBusinessName({
+        businessName: "my-cool-business",
+        taxId: "*****89000",
+      });
+      stubTaxEncryptionDecryptionClient.decryptValue.mockResolvedValue("123456789000");
+      apiTaxFilingClient.onboarding.mockResolvedValue(responseUserData);
+      const response = await request(app).post(`/onboarding`).send(taxIdAndBusinessName);
+      expect(response.body).toEqual(responseUserData);
+      expect(apiTaxFilingClient.onboarding).toHaveBeenCalledWith({
+        userData,
+        taxId: "123456789000",
+        businessName: "my-cool-business",
+      });
+      expect(stubUserDataClient.put).toHaveBeenCalledWith(responseUserData);
+      expect(stubUserDataClient.get).toHaveBeenCalledWith("some-id");
```

Comment:
there's an expectation missing that, once you add it, will reveal that this test isn't exactly set up to mirror the way it would look in the real world.

We should have an expectation to assert what the stubEncryptionDecrytionClient.decrypt was called with

Reply:
(same comment for the other mirrored test in lookup)